### PR TITLE
Better reporting for user limits

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Motivation and context

- Added explanatory messages for actions denied for user limits
- Fixed few rules and checks
- Upgraded OPA version

### How has this been tested?

Unit tests

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/iam/tests/test_rest_api.py [hunk 1]
```diff
@@ -20,6 +29,21 @@
...
 ]
 
+class ForceLogin:
+    def __init__(self, user, client):
+        self.user = user
+        self.client = client
+
+    def __enter__(self):
+        if self.user:
+            self.client.force_login(self.user, backend='django.contrib.auth.backends.ModelBackend')
```
Comment (nmanovic) [inline cvat/apps/iam/tests/test_rest_api.py:39]: Probably here we can configure huge limits for the user. What do you think? Let's add a function that writes big values for the user.

File: cvat/settings/testing.py [hunk 2]
```diff
@@ -64,6 +64,11 @@
     'django.contrib.auth.hashers.MD5PasswordHasher',
 )
 
+# Most of the tests don't suppose user limitations
+# When needed, this can be enabled with a decorator
+# and tested specifically
+IAM_USER_LIMITS_ENABLED = False
```
Comment (nmanovic) [inline cvat/settings/testing.py:70]: I don't like the approach. It is much better to have a configuration with default values for limits. Another approach is to use a function to update limits directly in old tests. For example, during login, the limits can be updated for a user.

File: cvat/apps/iam/permissions.py [hunk 3]
```diff
@@ -1138,20 +1347,266 @@ def get_common_data(db_job):
...
 
 
+class LimitPermission(OpenPolicyAgentPermission):
+    @classmethod
+    def create(cls, request, view, obj):
+        return [] # There are no basic (unconditional) permissions
+
+    @classmethod
+    def create_from_scopes(cls, request, view, obj, scopes: List[OpenPolicyAgentPermission]):
+        if not settings.IAM_USER_LIMITS_ENABLED:
```
Comment (nmanovic) [inline cvat/apps/iam/permissions.py:1357]: It is unacceptable to have such if in a production code. Let's remove it.

File: cvat/apps/limit_manager/core/limits.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            )
+
+        elif limit == Limits.ORG_TASKS:
+            assert context is not None
+            context = cast(OrgTasksContext, context)
+
+            return LimitStatus(
+                # TODO: check about active/removed tasks
+                Task.objects.filter(organization=context.org_id).count(),
+                10
```
Comment (nmanovic) [inline cvat/apps/limit_manager/core/limits.py:168]: @zhiltsov-max , better to have IAM_LIMITS in settings for such values now.

**Comment by zhiltsov-max** [comment]
@nmanovic , I addressed the comments.

File: cvat/apps/iam/rules/limits.rego [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+problems contains "user project tasks limit reached" if {
+    check_limit_exceeded(
+        input.resource.limits[CAP_TASKS_IN_USER_SANDBOX_PROJECT].used,
+        input.resource.limits[CAP_TASKS_IN_USER_SANDBOX_PROJECT].max
+    )
+}
+
+problems contains "org tasks limit reached" if {
```
Comment (nmanovic) [inline cvat/apps/iam/rules/limits.rego:51]: It is possible to use sprintf function to add information about current limits.

Comment (nmanovic) [reply cvat/apps/iam/rules/limits.rego:51]: sprintf("org projects limit reached. Current number of projects %d, maxim number of projects should be less than %d", [input.resource.limits[CAP_ORG_PROJECTS].used, input.resource.limits[CAP_ORG_PROJECTS].max])

File: cvat/apps/iam/rules/tests/configs/projects.csv [hunk 6]
```diff
@@ -1,10 +1,10 @@
 Scope,Resource,Context,Ownership,Limit,Method,URL,Privilege,Membership
-create,Project,Sandbox,N/A,resource['user']['num_resources'] < 3,POST,/projects,User,N/A
-create,Project,Organization,N/A,resource['user']['num_resources'] < 3,POST,/projects,User,Supervisor
+create,Project,Sandbox,N/A,,POST,/projects,User,N/A
+create,Project,Organization,N/A,,POST,/projects,User,Supervisor
 create,Project,Sandbox,N/A,,POST,/projects,Business,N/A
 create,Project,Organization,N/A,,POST,/projects,Business,Supervisor
```
Comment (kirill-sizov) [inline cvat/apps/iam/rules/tests/configs/projects.csv:5]: Now these rules for Business and for User are the same and so I think we should remove such repetition 
```suggestion
create,Project,Sandbox,N/A,,POST,/projects,User,N/A
create,Project,Organization,N/A,,POST,/projects,User,Supervisor
```
The same comment for all others csv files that have been changed in this PR

Comment (zhiltsov-max) [reply cvat/apps/iam/rules/tests/configs/projects.csv:5]: I'm not sure, the User and Business are still different privileges, aren't they?

Comment (kirill-sizov) [reply cvat/apps/iam/rules/tests/configs/projects.csv:5]: Yes, User and Business are different privileges, but these .csv files built by principle: each line means that a certain action is allowed for a user with certain privileges, as well as for all other users with higher privileges. 

And so if it was already mentioned that create,Project,Sandbox is allowed for User, this automatically means that this action is also allowed for Business. We also use this fact during test generation.

**Comment by nmanovic** [comment]
@zhiltsov-max , could you please resolve conflicts?

File: cvat/apps/iam/rules/tests/configs/projects.csv (hunk 6 above)
Comment (zhiltsov-max) [reply cvat/apps/iam/rules/tests/configs/projects.csv:5]: Ok, it makes sense then. I'll update the files.

Comment (zhiltsov-max) [reply cvat/apps/iam/rules/tests/configs/projects.csv]: Done

**Pull request merged**